Sage/carbon
# feat(loader,loader-bar): display loading message if user has opt-ed to reduced motion

### Proposed behaviour

- Alternative loading message for `Loader` and `LoaderBar` 
   - Replace animation with text when user has indicated a preference for reduced motion in their OS settings, or when the user's motion preference cannot be determined (e.g. with older browsers like IE11)
   - Add translation support for loading message (with English and Polish examples added to the respective built-in locales)
   - For `Switch`, remove fix width constraint to fit contents (specifically to support [loading pattern](https://carbon.sage.com/?path=/docs/switch--loading) since alternative text could be long)

#### `Loader` demo

https://user-images.githubusercontent.com/18368713/214578476-5b8f4bb2-910c-409e-9ad6-444c35a47a92.mov

#### `LoaderBar` demo

https://user-images.githubusercontent.com/18368713/214578532-c53f4f7e-15b8-421c-b251-2bf56603fc5c.mov

#### `Button` with `Loader` demo

https://user-images.githubusercontent.com/18368713/214578605-0eba7cb0-d7b1-4299-92ca-6be255500a99.mov

#### `Switch` with `Loader` demo

https://user-images.githubusercontent.com/18368713/214578667-59487ddf-146d-42ba-ac9a-e9bd6f268bab.mov

### Current behaviour

`Loader` and `LoaderBar`'s blinking and scrolling animations cannot be controlled by the user. Since these components could be used in parallel with other content, this could be distracting to users with reading difficulties or attention deficit disorders (see WCAG 2.1 Operable guideline ["Pause, Stop and Hide"](https://www.w3.org/TR/WCAG21/#pause-stop-hide) for details).

### Checklist




#### QA

 
### Testing instructions

- Open the following stories and with "Reduce Motion" enabled in the OS accessibility preferences, check that animation is replaced with the alternative loading message (for macOS, setting can be found in: `System Preferences > Accessibility > Display > Reduce Motion`):
    - `loader--default-story`
    - `loader-bar--default-story`
    - `switch--loading`
    - `select--with-is-loading-prop`
- Check visuals for loading message in the English and Polish locales, verify no regressions have been introduced.
